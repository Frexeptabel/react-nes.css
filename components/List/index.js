import React from "react";
import PropTypes from "prop-types";

const List = ({ listStyle = "disc", ...props }) => {
  <ul className={`nes-list is-${listStyle}`} {...props}>
    {props.children}
  </ul>;
};

List.propTypes = {
  listStyle: PropTypes.oneOf(["disc", "circle"])
};

export default List;
