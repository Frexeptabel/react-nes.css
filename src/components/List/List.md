List example:

```js
<List>
    <ul>List Item</ul>
    <ul>List Item</ul>
</List>
<List listStyle="circle">
    <ul>List Item</ul>
    <ul>List Item</ul>
</List>
```
