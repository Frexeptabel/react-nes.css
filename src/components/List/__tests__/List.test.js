import React from "react";
import List from "../index";
import renderer from "react-test-renderer";

test("The List should be rendered properly", () => {
  const component = renderer.create(
    <List>
      <ul>1</ul>
      <ul>2</ul>
    </List>
  );
  expect(component).toMatchSnapshot();
});

test("The List should be rendered properly", () => {
  const component = renderer.create(<List listStyle="circle" />);
  expect(component).toMatchSnapshot();
});
