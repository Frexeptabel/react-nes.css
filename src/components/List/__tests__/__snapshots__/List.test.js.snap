// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The List should be rendered properly 1`] = `
<ul
  className="nes-list is-disc"
>
  <ul>
    1
  </ul>
  <ul>
    2
  </ul>
</ul>
`;

exports[`The List should be rendered properly 2`] = `
<ul
  className="nes-list is-circle"
/>
`;
